staging/src/k8s.io/apimachinery/pkg/util/net/util.go

```diff
@@ -47,6 +48,11 @@ func IsConnectionReset(err error) bool {
 	return false
 }
 
+// Returns if the given err is "http2: client connection lost" error.
+func IsHTTP2ConnectionLost(err error) bool {
+	return err != nil && strings.Contains(err.Error(), "http2: client connection lost")
```

Comment:
if we're going to just match error strings, shouldn't we have a test that notices if the go string changes and breaks this method?

https://github.com/golang/net/blob/master/http2/transport.go#L1134

Reply:
I really wish we could be more robust here :-/

Reply:
@neild how do you feel about exposing the error in golang?

Reply:
> I really wish we could be more robust here :-/

hmm, the test setup would be painful to set up a http2 server and figure out a way to fail the ping health check. OTOH, I feel the possibility of change the error message in golang is extremely low.

Exposing the error in golang is much feasible

Reply:
> the test setup would be painful to set up a http2 server and figure out a way to fail the ping health check.

we do have that test
https://github.com/kubernetes/kubernetes/blob/bdbfbffef35c0ae46b9309fdab642f696e7a856a/staging/src/k8s.io/client-go/rest/connection_test.go#L87

Reply:
Thanks, it makes things easier. PTAL!